examples/rest-grpc-multiplex/src/main.rs

```diff
@@ -4,84 +4,90 @@
+        };
 
-// #[tonic::async_trait]
-// impl Greeter for GrpcServiceImpl {
-//     async fn say_hello(
-//         &self,
-//         request: tonic::Request<HelloRequest>,
-//     ) -> Result<TonicResponse<HelloReply>, Status> {
-//         tracing::info!("Got a request from {:?}", request.remote_addr());
+        Ok(TonicResponse::new(reply))
+    }
+}
 
-//         let reply = HelloReply {
-//             message: format!("Hello {}!", request.into_inner().name),
-//         };
+async fn web_root() -> &'static str {
+    tracing::info!("Got a REST request");
 
-//         Ok(TonicResponse::new(reply))
-//     }
-// }
+    "Hello, World!"
+}
 
-// async fn web_root() -> &'static str {
-//     "Hello, World!"
-// }
+#[tokio::main]
+async fn main() {
+    // initialize tracing
+    tracing_subscriber::registry()
+        .with(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| "example_rest_grpc_multiplex=debug".into()),
+        )
+        .with(tracing_subscriber::fmt::layer())
+        .init();
 
-// #[tokio::main]
-// async fn main() {
-//     // initialize tracing
-//     tracing_subscriber::registry()
-//         .with(
-//             tracing_subscriber::EnvFilter::try_from_default_env()
-//                 .unwrap_or_else(|_| "example_rest_grpc_multiplex=debug".into()),
-//         )
-//         .with(tracing_subscriber::fmt::layer())
-//         .init();
+    // build the rest service
+    let rest = Router::new().route("/", get(web_root));
 
-//     // build the rest service
-//     let rest = Router::new().route("/", get(web_root));
+    // build the grpc service
+    let reflection_service = tonic_reflection::server::Builder::configure()
+        .register_encoded_file_descriptor_set(proto::FILE_DESCRIPTOR_SET)
+        .build()
+        .unwrap();
 
-//     // build the grpc service
-//     let reflection_service = tonic_reflection::server::Builder::configure()
-//         .register_encoded_file_descriptor_set(proto::FILE_DESCRIPTOR_SET)
-//         .build()
-//         .unwrap();
-//     let grpc = tonic::transport::Server::builder()
-//         .add_service(reflection_service)
-//         .add_service(GreeterServer::new(GrpcServiceImpl::default()))
-//         .into_service();
+    let grpc = tonic::transport::Server::builder()
+        .add_service(reflection_service)
+        .add_service(GreeterServer::new(GrpcServiceImpl::default()))
+        .into_router();
 
-//     // combine them into one service
-//     let service = MultiplexService::new(rest, grpc);
+    // combine them into one service
+    let service = Steer::new(vec![rest, grpc], |req: &Request, _services: &[_]| {
+        if req
+            .headers()
+            .get(CONTENT_TYPE)
+            .map(|content_type| content_type.as_bytes())
+            .filter(|content_type| content_type.starts_with(b"application/grpc"))
+            .is_some()
```

Comment:
Nit, you could do just `.and_then(|content_type| content_type.as_bytes().starts_with(b"...").then_some(1)).unwrap_or(0)` without an explicit condition instead.

(Although now I see you have just copied this from the deleted file)